upgrade-os

```diff
@@ -0,0 +1,136 @@
+
+echo -e "\nStarting \033[1mlamassu-server\033[0m update. This will take a few minutes...\n"
+
+if [ "$(whoami)" != "root" ]; then
+  echo -e "This script has to be run as \033[1mroot\033[0m user"
+  exit 3
+fi
+
+# Use a lock file so failed scripts cannot be imediately retried
+# If not the backup created on this script would be replaced
+if ! mkdir /var/lock/lamassu-update; then
+  echo "Script is locked because of a failure." >&2
+  exit 1
+fi
+
+if [ -f /usr/bin/check-upgrade-status ]; then
+  rm -r /usr/bin/check-upgrade-status
+fi
+
+echo "tail -f ~/upgrade.log" > /usr/bin/check-upgrade-status
+sudo chmod +x /usr/bin/check-upgrade-status
+
+if [[ "$UBUNTU_VERSION" == "16.04" ]]; then
+  echo "Detected Ubuntu version: 16.04. Updating to Ubuntu version: 18.04..."
+  echo
+  echo -e "\033[0;33mNOTE: Some of the system's dependencies might require more reboots than expected during this process.\033[0m"
+  echo
+
+  command="/usr/bin/curl -sS https://raw.githubusercontent.com/lamassu/lamassu-install/electric-enlil/upgrade-os | bash >> ~/upgrade.log"
+  job="@reboot root sleep 5 | $command"
+  echo "$job" > /etc/cron.d/lamassu-upgrade
+
+  decho "updating to node 14"
+  npm install -g n
+  sudo n 14
+  sudo n which 14
+
+  checkDpkgLock
+  decho "updating to ubuntu 18.04"
+  sudo apt-get update -y --force-yes
+  checkIfCanContinue
+  sudo apt-get --with-new-pkgs upgrade -y --force-yes
+  checkIfCanContinue
+  sudo apt-get dist-upgrade -y --force-yes
+  checkIfCanContinue
+  sudo do-release-upgrade -f DistUpgradeViewNonInteractive
+
+  echo
+  echo -e "\033[0;33mUbuntu successfully updated to version 18.04!\033[0m"
+  echo -e "\033[0;33mThe machine will now reboot, resuming the update process right after.\033[0m"
+  echo -e "\033[0;33mDuring this time, paired machines will not be online.\033[0m"
+  echo
+  sudo shutdown -r now
+elif [[ "$UBUNTU_VERSION" == "18.04" ]]; then
+  echo "Detected Ubuntu version: 18.04. Updating to Ubuntu version: 20.04..."
+  echo
+  echo -e "\033[0;33mNOTE: Some of the system's dependencies might require more reboots than expected during this process.\033[0m"
+  echo
+
+  command="/usr/bin/curl -sS https://raw.githubusercontent.com/lamassu/lamassu-install/electric-enlil/upgrade-os | bash >> ~/upgrade.log"
+  job="@reboot root sleep 5 | $command"
+  echo "$job" > /etc/cron.d/lamassu-upgrade
+
+  checkDpkgLock
+  decho "updating to ubuntu 20.04"
+  sudo apt-get update -y --force-yes
+  checkIfCanContinue
+  sudo apt-get --with-new-pkgs upgrade -y --force-yes
+  checkIfCanContinue
+  sudo apt-get dist-upgrade -y --force-yes
+  checkIfCanContinue
+  sudo do-release-upgrade -f DistUpgradeViewNonInteractive
+
+  echo
+  echo -e "\033[0;33mUbuntu successfully updated to version 20.04!\033[0m"
+  echo -e "\033[0;33mThe machine will now reboot, allowing you to continue the update process right after.\033[0m"
+  echo -e "\033[0;33mDuring this time, paired machines will not be online.\033[0m"
+  echo
+  sudo shutdown -r now
```

Comment:
We need to `rm -r /var/lock/lamassu-update` before each instance of `shutdown -r now`, otherwise the `upgrade-os` script can't run upon logging back in.